Xamarin.Forms.Platform.UAP/ImageRenderer.cs

```diff
@@ -28,9 +29,28 @@ public override SizeRequest GetDesiredSize(double widthConstraint, double height
 
 			_measured = true;
 
-			var result = new Size { Width = ((BitmapSource)Control.Source).PixelWidth, Height = ((BitmapSource)Control.Source).PixelHeight };
-
-			return new SizeRequest(result);
+			if (Control.Source is BitmapSource bitmap)
+			{
+				return new SizeRequest(
+					new Size
+					{
+						Width = bitmap.PixelWidth,
+						Height = bitmap.PixelHeight
+					});
+			}
+			else if (Control.Source is CanvasImageSource canvas)
+			{
+				return new SizeRequest(
+					new Size
+					{
+						Width = canvas.SizeInPixels.Width,
+						Height = canvas.SizeInPixels.Height
+					});
+			}
+			else
+			{
+				throw new InvalidCastException($"\"{Control.Source.GetType().FullName}\" is not supported.");
+			}
```

Comment:
I would maybe put this into an extension method because I don't think this is the only place we might use this. I am thinking image buttons and other bits that use images.